src/mtcp/mtcp_restart.c

```diff
@@ -970,14 +975,20 @@ static void read_shared_memory_area_from_file(int fd, Area* area, int flags)
 
   /* Check to see if the filename ends with " (deleted)" */
   if (mtcp_strendswith(area_name, DELETED_FILE_SUFFIX)) {
+    // FIXME:  Shouldn't we also unlink area_name after re-mapping it?
     area_name[ mtcp_strlen(area_name) - mtcp_strlen(DELETED_FILE_SUFFIX) ] =
       '\0';
+    if (fix_parent_directories(area_name) != area_name) {
```

Comment:
Good point.  I'll remove this second call to fix_parent_directories() and add comments about the logic, for clarification.  You're two-for-two so far on catching my mistakes.  Thanks!

And in case we have to review this again, here was the previous logic:
  if MAP_SHARED:
    if ENOENT (open failed):
      if we can successfully fix parent directories:
        Go on and call read_shared_memory_area_from_file (in code below)
     else change MAP_SHARED to MAP_PRIVATE, and try again from the top:  goto read_data
   Still inside MAP_SHARED, read_shared_memory_area_from_file():
     fix_parent_directories (but it's redundant; we guaranteed the file does exist before going here)